compiledb-gen-parser

```diff
@@ -96,8 +100,26 @@ def parse_flags(build_log, proj_dir, inc_prefix, verbose):
     line_count = 0
     compiler = None
 
+    dir_stack = []
+    sub_dir = proj_dir
+
     # Process build log
     for line in build_log:
+        # Concatenate line if need
+        accumulate_line = line
+        while (line.endswith('\\\n')):
```

Comment:
Sorry, what case would this happen?

Reply:
@nickdiego Some makefiles contain lines ending with backslash and their build log contain backslash too. In this case the parser cannot produce a compilation db because a build command not present in one line. I encountered this condition when I work at a open source project named 'PJSIP', its build log as below:
```
for dir in pjlib/build pjlib-util/build pjnath/build third_party/build pjmedia/build pjsip/build pjsip-apps/build ; do \
                if /Applications/Xcode.app/Contents/Developer/usr/bin/make  -C $dir all; then \
                    true; \
                else \
                    exit 1; \
                fi; \
        done
/Applications/Xcode.app/Contents/Developer/usr/bin/make -f /Users/lihang/Projects/pjproject-2.6/build/rules.mak APP=PJLIB app=pjlib ../lib/libpj-x86_64-apple-darwin16.7.0.a
mkdir -p output/pjlib-x86_64-apple-darwin16.7.0
gcc -c -Wall -DPJ_AUTOCONF=1 -O2 -DPJ_IS_BIG_ENDIAN=0 -DPJ_IS_LITTLE_ENDIAN=1    -I../include \
                -o output/pjlib-x86_64-apple-darwin16.7.0/ioqueue_select.o \
                ../src/pj/ioqueue_select.c
gcc -c -Wall -DPJ_AUTOCONF=1 -O2 -DPJ_IS_BIG_ENDIAN=0 -DPJ_IS_LITTLE_ENDIAN=1    -I../include \
                -o output/pjlib-x86_64-apple-darwin16.7.0/file_access_unistd.o \
                ../src/pj/file_access_unistd.c
gcc -c -Wall -DPJ_AUTOCONF=1 -O2 -DPJ_IS_BIG_ENDIAN=0 -DPJ_IS_LITTLE_ENDIAN=1    -I../include \
                -o output/pjlib-x86_64-apple-darwin16.7.0/file_io_ansi.o \
                ../src/pj/file_io_ansi.c
gcc -c -Wall -DPJ_AUTOCONF=1 -O2 -DPJ_IS_BIG_ENDIAN=0 -DPJ_IS_LITTLE_ENDIAN=1    -I../include \
                -o output/pjlib-x86_64-apple-darwin16.7.0/os_core_unix.o \
                ../src/pj/os_core_unix.c
gcc -c -Wall -DPJ_AUTOCONF=1 -O2 -DPJ_IS_BIG_ENDIAN=0 -DPJ_IS_LITTLE_ENDIAN=1    -I../include \
                -o output/pjlib-x86_64-apple-darwin16.7.0/os_error_unix.o \
                ../src/pj/os_error_unix.c
gcc -c -Wall -DPJ_AUTOCONF=1 -O2 -DPJ_IS_BIG_ENDIAN=0 -DPJ_IS_LITTLE_ENDIAN=1    -I../include \
                -o output/pjlib-x86_64-apple-darwin16.7.0/os_time_unix.o \
                ../src/pj/os_time_unix.c
gcc -c -Wall -DPJ_AUTOCONF=1 -O2 -DPJ_IS_BIG_ENDIAN=0 -DPJ_IS_LITTLE_ENDIAN=1    -I../include \
                -o output/pjlib-x86_64-apple-darwin16.7.0/os_timestamp_posix.o \
                ../src/pj/os_timestamp_posix.c
```
Backslash is used very frequent in makefile so I put these codes into effect.

Thanks.

Reply:
Hmm, good catch! :)

So, just modify that var name to `working_dir`, and let's merge it.

Reply:
Done.